pkg/sql/pgwire/testdata/pgtest/copy

```diff
@@ -310,3 +310,184 @@ ReadyForQuery
+{"Type":"CommandComplete","CommandTag":"SELECT 2"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+# Verify that COPY CSV input can be split up at arbitrary points.
+send
+Query {"String": "DELETE FROM t"}
+Query {"String": "COPY t FROM STDIN WITH CSV"}
+CopyData {"Data": "1,o"}
+CopyData {"Data": "ne\n2,"}
+CopyData {"Data": "tw"}
+CopyData {"Data": "o\n"}
+CopyData {"Data": "3,\""}
+CopyData {"Data": "three"}
+CopyData {"Data": "\"\"e"}
+CopyData {"Data": "e\"\n4,\"four\n"}
+CopyData {"Data": "r\nr"}
+CopyData {"Data": "\"\n5,\"f\n\"\"ive\"\n6,\"six"}
+CopyData {"Data": "\""}
+CopyData {"Data": "\n7,seven\n8,eight\n9,\"n\nii"}
+CopyData {"Data": "ii\n"}
+CopyData {"Data": "ne\""}
+CopyDone
+Query {"String": "SELECT * FROM t ORDER BY i"}
+----
+
+until ignore=NoticeResponse ignore=RowDescription
+ReadyForQuery
+ReadyForQuery
+ReadyForQuery
+----
+{"Type":"CommandComplete","CommandTag":"DELETE 2"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+{"Type":"CopyInResponse","ColumnFormatCodes":[0,0]}
+{"Type":"CommandComplete","CommandTag":"COPY 9"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+{"Type":"DataRow","Values":[{"text":"1"},{"text":"one"}]}
+{"Type":"DataRow","Values":[{"text":"2"},{"text":"two"}]}
+{"Type":"DataRow","Values":[{"text":"3"},{"text":"three\"ee"}]}
+{"Type":"DataRow","Values":[{"text":"4"},{"binary":"666f75720a720a72"}]}
+{"Type":"DataRow","Values":[{"text":"5"},{"binary":"660a22697665"}]}
+{"Type":"DataRow","Values":[{"text":"6"},{"text":"six"}]}
+{"Type":"DataRow","Values":[{"text":"7"},{"text":"seven"}]}
+{"Type":"DataRow","Values":[{"text":"8"},{"text":"eight"}]}
+{"Type":"DataRow","Values":[{"text":"9"},{"binary":"6e0a696969690a6e65"}]}
+{"Type":"CommandComplete","CommandTag":"SELECT 9"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+# Verify that byte-escape sequences are processed in TEXT mode.
+# `MixceDU0` is base64 for `2,\x54`.
+send
+Query {"String": "DROP TABLE IF EXISTS tb"}
+Query {"String": "CREATE TABLE tb (i INT, b BYTEA)"}
+Query {"String": "COPY tb FROM STDIN DELIMITER ',' NULL ''"}
+CopyData {"Data": "1,one\n"}
+CopyData {"BinaryData": "MixceDU0"}
+CopyDone
+Query {"String": "SELECT i, encode(b, 'hex') FROM tb ORDER BY i"}
+----
+
+until ignore=RowDescription ignore=NoticeResponse
+ReadyForQuery
+ReadyForQuery
+ReadyForQuery
+ReadyForQuery
+----
+{"Type":"CommandComplete","CommandTag":"DROP TABLE"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+{"Type":"CommandComplete","CommandTag":"CREATE TABLE"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+{"Type":"CopyInResponse","ColumnFormatCodes":[0,0]}
+{"Type":"CommandComplete","CommandTag":"COPY 2"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+{"Type":"DataRow","Values":[{"text":"1"},{"text":"6f6e65"}]}
+{"Type":"DataRow","Values":[{"text":"2"},{"text":"54"}]}
+{"Type":"CommandComplete","CommandTag":"SELECT 2"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+# Verify that byte-escape sequences are processed in CSV mode.
+# `MixceDU0` is base64 for `2,\x54`.
```

Comment:
Does this support octal encoding as well like PostgreSQL [does](https://www.postgresql.org/docs/13/sql-copy.html)? Something like `1,"\157\156\145"`. Would be good to add a test if so.

Reply:
yes it should. adding a test